### PR TITLE
Issue/628 checkout agreements

### PR DESCRIPTION
**Pull request by mmularski**
### Description (*)

This PR converts the CreateTermEntityTest four variations to MFTF

### Fixed Issues (if relevant)

1. magento/magento-functional-tests-migration#628: Convert CreateTermEntityTest to MFTF

### Manual testing scenarios (*)

N/A

### CODE REVIEW

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermAbsentInCheckoutActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontAssertTermAbsentInCheckout">
+        <arguments>
+            <argument name="termCheckboxText" type="string"/>
+            <argument name="product" defaultValue="SimpleTwo"/>
+        </arguments>
+
+        <!--Add product to cart-->
```
Comment (lbajsarowicz) [inline app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermAbsentInCheckoutActionGroup.xml]: :up: for #720 and #719

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <fillField selector="{{CheckoutShippingSection.firstName}}" userInput="{{CustomerEntityOne.firstname}}" stepKey="enterFirstName"/>
+        <fillField selector="{{CheckoutShippingSection.lastName}}" userInput="{{CustomerEntityOne.lastname}}" stepKey="enterLastName"/>
+        <fillField selector="{{CheckoutShippingSection.street}}" userInput="{{CustomerAddressSimple.street[0]}}" stepKey="enterStreet"/>
+        <fillField selector="{{CheckoutShippingSection.city}}" userInput="{{CustomerAddressSimple.city}}" stepKey="enterCity"/>
+        <selectOption selector="{{CheckoutShippingSection.region}}" userInput="{{CustomerAddressSimple.state}}" stepKey="selectRegion"/>
+        <fillField selector="{{CheckoutShippingSection.postcode}}" userInput="{{CustomerAddressSimple.postcode}}" stepKey="enterPostcode"/>
+        <fillField selector="{{CheckoutShippingSection.telephone}}" userInput="{{CustomerAddressSimple.telephone}}" stepKey="enterTelephone"/>
+        <waitForLoadingMaskToDisappear stepKey="waitForLoadingMask1"/>
+        <click selector="{{CheckoutShippingMethodsSection.checkShippingMethodByName('')}}" stepKey="selectShippingMethod"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml]: :up: for #720 and #719

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        <see selector="{{StorefrontMessagesSection.success}}" userInput="You added {{product2.name}} to your shopping cart." stepKey="seeAddToCart2SuccessMessage"/>
+
+        <!--Go to Checkout Cart and proceed with multiple addresses-->
+        <amOnPage url="{{CheckoutCartPage.url}}" stepKey="goToCheckoutCart"/>
+        <waitForPageLoad stepKey="waitForPageLoad3"/>
+        <click selector="{{MultishippingSection.checkoutWithMultipleAddresses}}" stepKey="proceedMultishipping"/>
+
+        <!--Procees do overview page-->
+        <click selector="{{StorefrontMultishippingCheckoutAddressesToolbarSection.goToShippingInformation}}" stepKey="clickGoToShippingInformation"/>
+        <waitForPageLoad stepKey="waitForPageLoad4"/>
```
Comment (lbajsarowicz) [inline app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml]: That would be more readable if you name it `waitForShippingPageLoad` - for example.

File: app/code/Magento/CheckoutAgreements/Test/Mftf/Section/AdminNewTermFormSection.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<sections xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
+    <section name="AdminNewTermFormSection">
+        <element name="save" type="button" selector=".page-main-actions #save"/>
+
```
Comment (lbajsarowicz) [inline app/code/Magento/CheckoutAgreements/Test/Mftf/Section/AdminNewTermFormSection.xml]: :up: for #720 and #719 (Redundant empty line)

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontAssertTermInCheckout">
+        <arguments>
+            <argument name="termCheckboxText" type="string"/>
+            <argument name="product" defaultValue="SimpleTwo"/>
+        </arguments>
+
+        <!--Add product to cart-->
```
Comment (lbajsarowicz) [inline app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml]: :up: for #720 and #719

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontAssertTermRequireMessageInMultishippingCheckout">
+        <arguments>
+            <argument name="termCheckboxText" type="string"/>
+            <argument name="product1" defaultValue="SimpleTwo"/>
+            <argument name="product2" defaultValue="SimpleTwo"/>
+        </arguments>
+
+        <!--Add product1 to cart-->
```
Comment (lbajsarowicz) [inline app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml]: :up: for #720 and #719

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermAbsentInCheckoutActionGroup.xml (hunk 1 above)
Comment (mmularski) [reply app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermAbsentInCheckoutActionGroup.xml]: Fixed

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml (hunk 2 above)
Comment (mmularski) [reply app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml]: It should be empty

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml (hunk 3 above)
Comment (mmularski) [reply app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml]: Fixed - for all `waitForPageLoad` occurrences

File: app/code/Magento/CheckoutAgreements/Test/Mftf/Section/AdminNewTermFormSection.xml (hunk 4 above)
Comment (mmularski) [reply app/code/Magento/CheckoutAgreements/Test/Mftf/Section/AdminNewTermFormSection.xml]: Fixed

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml (hunk 5 above)
Comment (mmularski) [reply app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermInCheckoutActionGroup.xml]: Fixed

File: app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml (hunk 6 above)
Comment (mmularski) [reply app/code/Magento/CheckoutAgreements/Test/Mftf/ActionGroup/StorefrontAssertTermRequireMessageInMultishippingCheckoutActionGroup.xml]: Fixed

**Comment by mmularski** [comment]
@lbajsarowicz I fixed all related issues which you have requested. 
Let start with this PR. If everything will be OK, I will apply changes from this PR to #719 and #720

**Comment by tomreece** [comment]
I am processing this PR for delivery to mainline.
Please don't make any additional changes at this time because those commits may be missed and not merged correctly.

**Pull request merged**